src/borg/helpers.py

```diff
@@ -722,17 +722,7 @@ def format_archive(archive):
     )
 
 
-def memoize(function):
-    cache = {}
-
-    def decorated_function(*args):
-        try:
-            return cache[args]
-        except KeyError:
-            val = function(*args)
-            cache[args] = val
-            return val
-    return decorated_function
+memoize = lru_cache(maxsize=None)
```

Comment:
maybe go one step further and replace memoize usages by @lrucache?
also the maxsize=None should be checked then, whether it makes sense or can be harmful.